exercicios/para-casa/entregas/exemplo-nome-sobrenome/jeanne-mendes/projeto/Conta/Conta.js

```diff
@@ -0,0 +1,183 @@
+class Conta {
+  #agencia;
+  #conta;
+  #saldo;
+  chavesPix;
+  static listaContas = [];
+
+  constructor(agencia, conta, saldo) {
+    this.#agencia = agencia;
+    this.#conta = conta;
+    this.#saldo = saldo;
+    this.chavesPix = {
+      cpf: undefined,
+      email: undefined,
+      telefone: undefined,
+    };
+
+    Conta.listaContas.push(this);
+  }
+
+  //metodo para destruir objeto da lista de contas pra salvar sua memoria
+  destruir() {
+    let i = Conta.listaContas.indexOf(this);
+    Conta.listaContas.splice(i, 1);
+  }
+
+  criarConta(agencia, conta, saldo) {
+    if (agencia.length === 4 && conta.length === 5 && saldo > 0) {
+      this.#agencia = agencia;
+      this.#conta = conta;
+      this.#saldo = saldo;
+
+      return "Conta criada com sucesso";
+    } else {
+      throw new Error("Dados inválidos para cadastro");
+    }
+  }
+  sacar(valor) {
+    if (valor > 0 && typeof valor === "number") {
+      if (this.#saldo - valor > 0) {
+        const saldoAtualizado = this.#saldo - valor;
+        this.setSaldo(saldoAtualizado);
+      } else {
+        throw new Error("Saldo insuficiente");
+      }
+    } else {
+      throw new Error("Valor inválido para saque");
+    }
+  }
+
+  depositar(valor) {
+    if (valor > 0 && typeof valor === "number") {
+      const saldoAtualizado = this.#saldo + valor;
+      this.setSaldo(saldoAtualizado);
+    } else {
+      throw new Error("Valor inválido para depósito");
+    }
+  }
+
+  transferir(valor, agencia, conta) {
+    //LISTA.find(APELIDO PARA ITEM SELECIONADA => COMPARACAO )
+    /**
+     * antes de fazer a transferencia preciso verificar se a conta receptora existe na lista de contas
+     * contaValida vai me retornar a conta se ela existir e undefined se não existir
+     */
+    let contaValida = Conta.listaContas.find((contaReceptora) => {
+      let numeroContaReceptora = contaReceptora.getConta();
+      let numeroAgenciaReceptora = contaReceptora.getAgencia();
+      return (
+        numeroContaReceptora === conta && numeroAgenciaReceptora === agencia
+      );
+    });
+
+    if (!contaValida) {
+      throw new Error("Conta não encontrada");
+    }
+
+    if (valor < 0) {
+      throw new Error("Valor inválido para transferencia");
+    }
+
+    //a conta não pode ficar negativa ao fazer a transferencia,
+    if (this.#saldo - valor > 0) {
+      const saldoAtualizado = this.#saldo - valor;
+      this.setSaldo(saldoAtualizado);
+      const saldoContaReceptora = contaValida.getSaldo() + valor;
+      contaValida.setSaldo(saldoContaReceptora);
+      return "Tranferencia realizada";
+    }
+
+    if (valor > this.#saldo) {
```

Comment:
Você não acha que esse if poderia ser apagado já que la em cima estamos fazendo `if (this.#saldo - valor > 0)` e esse código por si só faz o mesmo que esse, já que se valor for maior que saldo ele será menor que zero.